crypto/fipsmodule/ec/ec.c

```diff
@@ -506,17 +488,20 @@ int EC_GROUP_cmp(const EC_GROUP *a, const EC_GROUP *b, BN_CTX *ignored) {
     return 0;
   }
 
-  // |a| and |b| are both custom curves. We compare the entire curve
-  // structure. If |a| or |b| is incomplete (due to legacy OpenSSL mistakes,
-  // custom curve construction is sadly done in two parts) but otherwise not the
-  // same object, we consider them always unequal.
-  return a->meth != b->meth ||  //
-         !a->has_order || !b->has_order ||
-         BN_cmp(&a->order.N, &b->order.N) != 0 ||
-         BN_cmp(&a->field.N, &b->field.N) != 0 ||
-         !ec_felem_equal(a, &a->a, &b->a) ||  //
-         !ec_felem_equal(a, &a->b, &b->b) ||
-         !ec_GFp_simple_points_equal(a, &a->generator.raw, &b->generator.raw);
+  // |a| and |b| are both custom curves. We compare the entire curve structure
+  // if both |a| and |b| are complete. If either are incomplete (due to legacy
+  // OpenSSL mistakes, custom curve construction is sadly done in two parts), we
+  // only compare the parts that are available.
+  if (a->has_order && b->has_order) {
+    return a->meth != b->meth || BN_cmp(&a->order.N, &b->order.N) != 0 ||
+           BN_cmp(&a->field.N, &b->field.N) != 0 ||
+           !ec_felem_equal(a, &a->a, &b->a) ||
+           !ec_felem_equal(a, &a->b, &b->b) ||
+           !ec_GFp_simple_points_equal(a, &a->generator.raw, &b->generator.raw);
```

Comment:
Great question, I assumed so since we had already been doing this and the [criteria](https://github.com/aws/aws-lc/blob/bda01b40727126e8cc2ec1e375301700b0fa6ba7/crypto/fipsmodule/ec/simple.c#L224-L269) of `ec_GFp_simple_points_equal` was relatively strict as opposed to  `ec_felem_equal`. But I'm hesitant on jumping to my own conclusions on this, I'd feel more comfortable getting someone more familiar with elliptic curves to weigh in. 

cc @dkostic: Do we need an additional check for `ec_GFp_simple_points_equal(b, &a->generator.raw, &b->generator.raw)` at this point?

Reply:
we don't need the additional check at this point